light-clients/icsxx-cf-solana/src/header.rs

```diff
@@ -1,2 +1,397 @@
-super::wrap!(cf_guest_upstream::Header<PK> as Header);
-super::wrap!(impl<PK> proto for Header);
+use crate::{
+	proto,
+	solana::{
+		blockstore::{get_completed_data_ranges, get_slot_entries_in_block},
+		shred::{shred_code::ShredCode, shred_data::ShredData, *},
+	},
+};
+use alloc::vec::Vec;
+use guestchain::PubKey;
+use proto_utils::BadMessage;
+use solana_sdk::{clock::Slot, hash::Hash, signature::Signature};
+use std::{collections::BTreeSet, convert::From, marker::PhantomData};
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Header<PK> {
```

Comment:
If the header doesn’t depend on `PK` let just drop it?

Reply:
I thought maybe you'll use it for your part of work, but if you think it's not needed I'll remove it

Reply:
PK would be used for places where signature is stored. This code uses Signature from solana_sdk so I don’t think there is a need for a generic.